src/hdcontroller.ts

```diff
@@ -298,14 +298,16 @@ export class HDProxydClient {
     public put(stream: Stream, size: number, params: any, reqUids: string, callback: HDProxydClientPutCallback): void {
         const log = this.createLogger(reqUids);
         this._failover('POST', stream, size, '', 0, log, (err?: HDProxydError, response?: http.IncomingMessage) => {
+            if (response) {
+                response.resume();
+            }
```

Comment:
I dont understand why moving the `response.resume()` here solves the issue.

Basically, if there is a response of the request, we force the stream to continue to flow so we can read the data? The `end` event of the stream will come later and trigger a close?

Reply:
Maybe we need to check the err too?

```
if (err && response) {
    response.resume();
}

Reply:
Good catch, it does not fix the issue.
It was intended when/if we have both an error and a response in the same call.

In the issue at hands, it is not sufficient.
I am considering letting this code (it is not harmful and is doing the right thing IMO), though

Reply:
OK for keeping this code